packages/smooth_app/lib/helpers/product_cards_helper.dart

```diff
@@ -96,31 +96,31 @@ List<ProductImageData> getProductMainImagesData(
     <ProductImageData>[
       ProductImageData(
         imageField: ImageField.FRONT,
-        imageUrl: product.imageFrontUrl,
+        imageDescriptor: ImageDescriptor.fromUrl(product.imageFrontUrl),
         title: appLocalizations.product,
         buttonText: appLocalizations.front_photo,
       ),
       ProductImageData(
         imageField: ImageField.INGREDIENTS,
-        imageUrl: product.imageIngredientsUrl,
+        imageDescriptor: ImageDescriptor.fromUrl(product.imageIngredientsUrl),
         title: appLocalizations.ingredients,
         buttonText: appLocalizations.ingredients_photo,
       ),
       ProductImageData(
         imageField: ImageField.NUTRITION,
-        imageUrl: product.imageNutritionUrl,
+        imageDescriptor: ImageDescriptor.fromUrl(product.imageNutritionUrl),
```

Comment:
This is a typical case where I don't like your `ImageDescriptor`: that's internally that `ProductImageData` should deal with it, like a black box. What's the added value of always having to add `ImageDescriptor.from(...)` in all constructor calls: just do it inside your constructor if you need it.